sql/spanner/inspect.go

```diff
@@ -0,0 +1,738 @@
+	var schemas []string
+	for _, s := range realm.Schemas {
+		sName := s.Name
+		// Here we reverse the schema alias.
+		if s.Name == defaultSchemaNameAlias {
+			sName = ""
+		}
+		schemas = append(schemas, sName)
+	}
+	rows, err := i.QueryContext(ctx, tablesQuery, schemas)
+	if err != nil {
+		return fmt.Errorf("query tables: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var tSchema, name, parentTable, onDeleteAction, spannerState sql.NullString
+		if err := rows.Scan(&tSchema, &name, &parentTable, &onDeleteAction, &spannerState); err != nil {
+			return fmt.Errorf("scan table information: %w", err)
+		}
+		if !sqlx.ValidString(name) {
+			return fmt.Errorf("invalid table name: %q", name.String)
+		}
+		sName := tSchema.String
+		if sName == "" {
+			sName = defaultSchemaNameAlias
+		}
+		s, ok := realm.Schema(sName)
+		if !ok {
+			return fmt.Errorf("schema %q was not found in realm", tSchema.String)
+		}
+		t := &schema.Table{
+			Name: name.String,
+		}
+		s.AddTables(t)
+		// TODO(tmc): handle parentTable, onDeleteAction, spannerState as attrs
+	}
+	if rows.Err() != nil {
+		return rows.Err()
+	}
+	return rows.Close()
+}
+
+// columns queries and appends the columns of the given table.
+func (i *inspect) columns(ctx context.Context, s *schema.Schema) error {
+	query := columnsQuery
+	rows, err := i.querySchema(ctx, query, s)
+	if err != nil {
+		return fmt.Errorf("spanner: querying schema %q columns: %w", s.Name, err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		if err := i.addColumn(s, rows); err != nil {
+			return fmt.Errorf("spanner: %w", err)
+		}
+	}
+	if err := rows.Err(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// addColumn scans the current row and adds a new column from it to the table.
+func (i *inspect) addColumn(s *schema.Schema, rows *sql.Rows) error {
+	var (
+		table, column, spannerType, colDefault sql.NullString
+		genExpr, spannerState                  sql.NullString
+		nullable, stored, generated            sql.NullBool
+		ord                                    sql.NullInt64
+
+		err error
+	)
+	if err := rows.Scan(
+		&table, &column, &ord, &colDefault,
+		&nullable, &spannerType, &generated, &genExpr,
+		&stored, &spannerState,
+	); err != nil {
+		return err
+	}
+	t, ok := s.Table(table.String)
+	if !ok {
+		return fmt.Errorf("table %q was not found in schema", table.String)
+	}
+	c := &schema.Column{
+		Name: column.String,
+		Type: &schema.ColumnType{
+			Raw:  spannerType.String,
+			Null: nullable.Bool,
+		},
+	}
+
+	// Converts spanner string type to schema.Type.
+	c.Type.Type, err = columnType(spannerType.String)
+	if err != nil {
+		return fmt.Errorf("spanner: Unable to convert string %q to schema.Type: %w", spannerType.String, err)
+	}
+
+	if colDefault.Valid {
+		c.Default = defaultExpr(c, colDefault.String)
+	}
+	if generated.Bool {
+		c.Attrs = []schema.Attr{
+			&schema.GeneratedExpr{
+				Expr: genExpr.String,
+				Type: "STORED",
```

Comment:
```suggestion
				Type: stored,
```
Use the const defined in `driver.go`.